libs/agno/agno/workflow/workflow.py

```diff
@@ -2524,7 +2524,7 @@ async def arun(
 
         # Read existing session from database
         if self._has_async_db():
-            workflow_session = await self.aread_or_create_session(session_id=session_id, user_id=user_id)
+            workflow_session = asyncio.run(self.aread_or_create_session(session_id=session_id, user_id=user_id))
```

Comment:
No we'll have to pass this step further down the chain. See how we did it for agents/teams